frontend/src/metabase/static-viz/components/XYChart/XYChart.tsx

```diff
@@ -162,11 +169,16 @@ export const XYChart = ({
   const areXTicksHidden = settings.x.tick_display === "hide";
   const xLabelOffset = areXTicksHidden ? -style.axes.ticks.fontSize : undefined;
 
-  const tickValues = fixTimeseriesTicksExceedXTickCount(
-    settings.x.type,
-    xScale.scale,
-    xTickCount,
-  );
+  useEffect(() => {
+    setXTickValues(
```

Comment:
I now put `xTickValues` inside a state, so I need to make all their dependencies memoized.

Reply:
AFAIK this code would not run in the static viz SSR 🤔

Reply:
That's interesting. I'll test it to see how it goes.

Reply:
You are absolutely right again. This doesn't work on SSR, I'm trying to see what I can do about this, because the whole change is to utilize state to allow consistent left padding 🤔